lib/libs/sink-lib.ts

```diff
@@ -90,21 +90,22 @@ const prettyPrintJsonInObject = (obj: any): any => {
 export function getDomainAndNamespace<T extends BaseIndex>(
   baseIndex: T,
 ): { domain: string; index: `${string}${T}` };
-export function getDomainAndNamespace(): { domain: string; index: string };
+
 export function getDomainAndNamespace(baseIndex?: BaseIndex) {
   const domain = process.env.osDomain;
 
   if (domain === undefined) {
     throw new Error("osDomain is undefined in environment variables");
   }
 
-  const indexNamespace = process.env.indexNamespace;
-
-  if (indexNamespace === undefined) {
+  if (indexNamespace !== "" && indexNamespace === undefined) {
     throw new Error("indexName is undefined in environment variables");
```

Comment:
Might wanna adjust the error message to include something about a missing string. Also, I might be reading this wrong, but is `indexNamespace` accessed before init on line 105?
```suggestion
    throw new Error("indexName is either undefined or missing a value in environment variables");
```